### PR TITLE
KIE forms jBPM Integration

### PR DESCRIPTION
**Pull request by pefernan**
- Added Providers for forms.

### CODE REVIEW

**Comment by pefernan** [comment]
@mswiderski can you take a look?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
+package org.kie.server.services.jbpm.ui.form;
+
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.Map;
+import java.util.stream.Collectors;
+import javax.enterprise.context.Dependent;
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java]: could we avoid CDI annotations in kie server? and I guess Dependent is default if not defined, right?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteFormModellerFormProvider.java [hunk 2]
```diff
@@ -43,7 +43,7 @@
 import org.w3c.dom.Node;
 import org.w3c.dom.NodeList;
 
-public class InMemoryFormProvider extends AbstractFormProvider implements UIFormProvider {
+public class RemoteFormModellerFormProvider extends AbstractFormProvider implements UIFormProvider {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteFormModellerFormProvider.java]: what's the point of renaming class here?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package org.kie.server.services.jbpm.ui.form;
+
+import org.jbpm.kie.services.impl.FormManagerService;
+import org.jbpm.kie.services.impl.form.provider.InMemoryFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+public class RemoteClassPathFormProvider extends InMemoryFormProvider implements UIFormProvider {
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java]: isn't the InMemoryFormProvider class was renamed?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider [hunk 4]
```diff
@@ -1 +1,4 @@
-org.kie.server.services.jbpm.ui.form.InMemoryFormProvider
\ No newline at end of file
+org.kie.server.services.jbpm.ui.form.RemoteFormModellerFormProvider
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider]: why do we have so many providers?

**Comment by pefernan** [comment]
@mswiderski I added new providers to handle .ftl template forms. That's why I also renamed the existing provider, to match the existing provider names on jBPM with the ones here.

BTW, tests & licenses are comming, sorry for that.

**Comment by pefernan** [comment]
@mswiderski added missing license headers & added all formtypes to test that the providing engine covers all form types.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+
+package org.kie.server.services.jbpm.ui.form;
+
+import javax.enterprise.context.Dependent;
+
+import org.jbpm.kie.services.impl.FormManagerService;
+import org.jbpm.kie.services.impl.form.provider.InMemoryFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteClassPathFormProvider.java]: can we remove this?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteInMemoryFormProvider.java [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+package org.kie.server.services.jbpm.ui.form;
+
+import javax.enterprise.context.Dependent;
+
+import org.jbpm.kie.services.impl.FormManagerService;
+import org.jbpm.kie.services.impl.form.provider.ClasspathFormProvider;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteInMemoryFormProvider.java]: same here, can we remove the @Dependent annotation?

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java [hunk 7]
```diff
@@ -0,0 +1,156 @@
...
+import com.google.gson.JsonParser;
+import org.apache.commons.lang3.StringUtils;
+import org.jbpm.kie.services.impl.FormManagerService;
+import org.jbpm.kie.services.impl.form.provider.AbstractFormProvider;
+import org.jbpm.kie.services.impl.model.ProcessAssetDesc;
+import org.jbpm.services.api.model.ProcessDefinition;
+import org.kie.api.task.model.Task;
+import org.kie.server.services.jbpm.ui.api.UIFormProvider;
+
+@Dependent
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/java/org/kie/server/services/jbpm/ui/form/RemoteKieFormsProvider.java]: and here as well, can we remove it?

**Comment by mswiderski** [comment]
@pefernan so we don't support .form (old form modeler format) anymore, right?

**Comment by pefernan** [comment]
@mswiderski we do still support the old .form format. I haven't removed the old form engine yet, we still must decide what to do about it.

BTW: removed annotations.

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
@mswiderski can you take a look another time please??

**Comment by mswiderski** [comment]
jenkins retest this

**Comment by mswiderski** [comment]
changes look good, nice work! let's wait for jenkins to rerun the build.

File: kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider [hunk 8]
```diff
@@ -1 +1,4 @@
...
+org.kie.server.services.jbpm.ui.form.RemoteKieFormsProvider
+org.kie.server.services.jbpm.ui.form.RemoteInMemoryFormProvider
+org.kie.server.services.jbpm.ui.form.RemoteClassPathFormProvider
```
Comment (mswiderski) [inline kie-server-parent/kie-server-services/kie-server-services-jbpm-ui/src/main/resources/META-INF/services/org.kie.server.services.jbpm.ui.api.UIFormProvider]: RemoteClassPathFormProvider does not exists any more and that's why build is failing:

Caused by: java.util.ServiceConfigurationError: org.kie.server.services.jbpm.ui.api.UIFormProvider: Provider org.kie.server.services.jbpm.ui.form.RemoteClassPathFormProvider not found

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
jenkins retest this.

**Comment by pefernan** [comment]
jenkins retest this

**Pull request merged**